src/marketaggregator.ts

```diff
@@ -105,6 +109,11 @@ export class MarketAggregator {
       const xStepDataMap = await this.xStep.query(stepMarketData.price);
       markets = { ...markets, ...xStepDataMap };
     }
+    const invictusMarketData = coingeckoMarketDataMap[INVICTUS_MINT];
```

Comment:
Probably not a change for now as it also affects xSTEP - but I think it would be better if we abstract away the mint look-ups and perhaps pass in the market map to the `query` function instead of the price. I worry that `MargetAggregator` could get bloated with a lot of imports it probably shouldn't care about. @ebramanti keen to hear your thoughts too.

Reply:
Don't think this has anything to do with imports, this data is loaded in the current function. However, could be a bit cleaner to encapsulate this behavior inside of the respective market sources. We should return an empty object if we are unable to find a price for the relative asset (in this case, $STEP and $IN).

Reply:
The way to do it would be to pass the `markets` variable as an argument, and have these market sources accept a `MarketDataMap` type as an argument to the query function.

Reply:
Yep that's what I was suggesting above :+1: 

The other thing we should do is move away from individual instances and simply have an array of `MarketSource` since they all have the same `query` interface we can just loop through and pass in `MarketDataMap`.